[WPE] WPE Platform: convert WPEToplevel get_max_views vfunc into a construct only property

carlosgcampos opened the pull request:
#### a2d7a9a4769db9936912bf0600bd34b096724ac0
<pre>
[WPE] WPE Platform: convert WPEToplevel get_max_views vfunc into a construct only property
<a href="https://bugs.webkit.org/show_bug.cgi?id=291080">https://bugs.webkit.org/show_bug.cgi?id=291080</a>

Reviewed by Patrick Griffis.

Because not only the platform needs to support the multiple views, but
also the applications. When applications want to handle the toplevel
themselves they need to decide the maximum view allowed.

* Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp:
(wpeToplevelSetProperty):
(wpeToplevelGetProperty):
(wpe_toplevel_class_init):
(wpe_toplevel_get_max_views):
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_set_toplevel):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandCreateView):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(wpe_toplevel_wayland_new):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.h:

Canonical link: <a href="https://commits.webkit.org/293405@main">https://commits.webkit.org/293405@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4401b139229e7d2dcb693377ae950f01c05c2bff

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98847 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18484 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8725 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103973 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49436 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18779 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26934 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/75253 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32384 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101851 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14268 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89271 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55612 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/14059 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48816 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/83998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7314 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106342 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25944 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/84217 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26319 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85469 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83715 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28366 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/6036 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19658 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16064 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25897 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/31083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25717 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29037 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27291 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

csaavedra commented on Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp:
```diff
@@ -154,6 +162,19 @@ static void wpe_toplevel_class_init(WPEToplevelClass* toplevelClass)
             WPE_TYPE_DISPLAY,
             static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
 
+    /**
+     * WPEToplevel:max-views:
+     *
+     * The maximum number of #WPEView that can be added to the #WPEToplevel.
+     * A value of 0 means no limit.
+     */
```
Shouln't this have a Since tag?

carlosgcampos replied:
No, because we haven't released this api yet.

carlosgcampos committed:
[WPE] WPE Platform: convert WPEToplevel get_max_views vfunc into a construct only property
https://bugs.webkit.org/show_bug.cgi?id=291080

Reviewed by Patrick Griffis.

Because not only the platform needs to support the multiple views, but
also the applications. When applications want to handle the toplevel
themselves they need to decide the maximum view allowed.

* Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp:
(wpeToplevelSetProperty):
(wpeToplevelGetProperty):
(wpe_toplevel_class_init):
(wpe_toplevel_get_max_views):
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.h:
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpe_view_set_toplevel):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandCreateView):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(wpe_toplevel_wayland_new):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.h:

Canonical link: https://commits.webkit.org/293405@main
